file_link/hardlink.py

```diff
@@ -0,0 +1,43 @@
+import platform
+
+
+def create_windows(source, destination):
+    """Creates hardlink at destination from source in Windows."""
+
+    import ctypes
+    from ctypes import WinError
+    from ctypes.wintypes import BOOL
+    CreateHardLink = ctypes.windll.kernel32.CreateHardLinkW
+    CreateHardLink.argtypes = [
+        ctypes.c_wchar_p, ctypes.c_wchar_p, ctypes.c_void_p
+    ]
+    CreateHardLink.restype = BOOL
+
+    res = CreateHardLink(destination, source, None)
+    if res == 0:
+        raise WinError()
+
+
+def create_linux(source, destination):
+    """Creates hardlink at destination from source in Linux."""
+    raise NotImplementedError("Linux is not support yet.")
+
+
+def create_osx(source, destination):
+    """Creates hardlink at destination from source in OSX."""
+    raise NotImplementedError("OSX is not support yet.")
+
+
+def create(source, destination):
```

Comment:
I would call this simply `link` and make the arguments `source` and `link_name` so we align with [`os.link`](https://docs.python.org/2/library/os.html#os.link). It's not the prettiest argument names, but better to align with something than nothing.